chapters/packages.tex

```diff
@@ -326,3 +326,123 @@ \section{External resources}\label{external-resources}
+It contains all necessary information to translate all descriptions, but no translations.
+The pattern \filename{<LibraryName>} denotes the toplevel class name of the library.
+\item One file for each supported language with the name \filename{<LibraryName>.<language>.po}, as the resource \filename{modelica://<LibraryName>/Resources/Language/<LibraryName>.<language>.po}.
+This file is a copy of the associated template file, but extended with the translations in the specified language.
+The pattern \filename{<language>} stands for the ISO 639-1 language code, e.g. \filename{de} or \filename{en}.
+\end{enumerate}
+
+The files consist of a header and a body.
+The detailed format of these files is described in \href{https://www.gnu.org/software/gettext/manual/gettext.pdf}{GNU gettext}.
+\begin{nonnormative}
+All text strings are in double quotes and encoded with UTF-8 characters.
+Comments start with an \lstinline!#! and are continued until the end of line.
+Spaces outside strings are ignored and used as separators.
+
+The header is marked with an empty \emph{msgid} entry and looks like this:
+\begin{lstlisting}
+# SOME DESCRIPTIVE TITLE.
+# Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER
+# This file is distributed under the same license as the PACKAGE package.
+# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
+#
+#, fuzzy
+msgid ""
+msgstr ""
+"Project-Id-Version: PACKAGE VERSION\n"
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2019-03-15 10:52+0100\n"
+"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+"Language-Team: LANGUAGE <LL@li.org>\n"
+"Language: \n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+\end{lstlisting}
+All general terms in the header should be replaced by specific information.
+
+In the body, directly following the header, for each descriptive text there is an entry of the form
+\begin{lstlisting}
+#: <file name>:<line number>
+msgctxt "<context identifier>"
+msgid "<descriptive text>"
+msgstr "<translation>"
+\end{lstlisting}
+\end{nonnormative}
+Only the keywords \lstinline!msgctxt!, \lstinline!msgid! and \lstinline!msgstr! are used.
+
+\begin{nonnormative}
+At first there can be an optional comment describing the location (file name and line number) of the text to translate.
+Multiple occurences of the same string can be listed here, separated by space.
+\end{nonnormative}
+
+The \lstinline!<context identifier>! following the keyword \lstinline!msgctxt! shall be the full name of the Modelica class (e.g., \lstinline!Modelica.Blocks.Math.Sin!) where the text appears.
+Short class definitions do not appear here.
+Texts in such classes belong to the enclosing full class definition.
+
+\begin{nonnormative}
+After the \lstinline!msgid! keyword the text string which shall be translated follows.
+\end{nonnormative}
+It should contain the original string from the Modelica text representation.
```

Comment:
I think these two sentences need to be combined into a single normative paragraph; otherwise the second sentence has a strange reference out of the normative text.
While this sentence just expresses a fact about gettext, I think that the following sentence depends upon this for context, and that both sentences therefore need to be in the same normative paragraph.
```suggestion
The text string which shall be translated (following the \lstinline!msgid! keyword) shall contain the original string from the Modelica code.
```

(I also noted that we don't use the terminology _Modelica text representation_.)